examples/tinytest/src/main.rs

```diff
@@ -33,12 +33,13 @@ fn save_file_dialog(title: &CStr, path: &CStr,
                     num_patterns: c_int,
                     filter_pattern_one: &CStr, des: &CStr) -> *const c_char {
     let filter_patterns : [&CStr; 1] = [filter_pattern_one];
+    let ptr_filter_patterns = filter_patterns.iter().map(|c| c.as_ptr()).collect::<Vec<*const c_char>>().as_ptr();
```

Comment:
Sorry, this is still unsafe. The as_ptr() is what makes the Vec go out of scope; it should be moved to the actual argument to the function call instead.